### PR TITLE
Add support for storing OpenDroneID UAS ID in persistent storage

### PR DESCRIPTION
**Pull request by bugobliterator**
* Also allows for using Unique ID as the remaining part of the UAS ID.

The UAS ID is set per ANSI/CTA-2063-A

### CODE REVIEW

File: libraries/AP_OpenDroneID/AP_OpenDroneID.cpp [hunk 1]
```diff
@@ -95,7 +103,81 @@ void AP_OpenDroneID::init()
...
+        // prepare basic id pkt
+        AP_Param::get("SYSID_THISMAV", val);
+        pkt_basic_id.target_system = val;
+        pkt_basic_id.target_component = MAV_COMP_ID_ODID_TXRX_1;
+        pkt_basic_id.id_type = MAV_ODID_ID_TYPE_SERIAL_NUMBER;
+        pkt_basic_id.ua_type = OPENDRONEID_UA_TYPE;
+
+        uint8_t sys_id[12];
+        uint8_t len = 12;
+        hal.util->get_system_id_unformatted(sys_id, len);
```
Comment (tridge) [inline libraries/AP_OpenDroneID/AP_OpenDroneID.cpp:148]: this assumes stm32 ID layout, not true on linux hal or esp32, or SITL
maybe use crc_crc64() ?

Comment (tridge) [reply libraries/AP_OpenDroneID/AP_OpenDroneID.cpp:148]: or remove feature is fine too

File: libraries/AP_OpenDroneID/AP_OpenDroneID.cpp [hunk 2]
```diff
@@ -95,7 +103,81 @@ void AP_OpenDroneID::init()
...
+        snprintf(buffer, sizeof(buffer), "%sE%x%x%x%x%x%x%x",
+                        OPENDRONEID_UA_MFR_CODE,
+                        sys_id[0], sys_id[1], sys_id[2],
+                        sys_id[4],
+                        sys_id[5], sys_id[6], sys_id[7]);
+        memcpy(pkt_basic_id.uas_id, buffer, sizeof(pkt_basic_id.uas_id));
+} else 
+#endif
+    if (id_len > 0) {
+        float val;
```
Comment (tridge) [inline libraries/AP_OpenDroneID/AP_OpenDroneID.cpp:160]: = 0

File: libraries/AP_OpenDroneID/AP_OpenDroneID.cpp [hunk 3]
```diff
@@ -95,7 +103,81 @@ void AP_OpenDroneID::init()
...
+        // prepare basic id pkt
+        AP_Param::get("SYSID_THISMAV", val);
```
Comment (tridge) [inline libraries/AP_OpenDroneID/AP_OpenDroneID.cpp:162]: can do gcs().sysid_this_mav()

File: libraries/AP_HAL_ChibiOS/Util.cpp [hunk 4]
```diff
@@ -560,6 +568,37 @@ bool Util::load_persistent_params(ExpandingString &str) const
...
+    if (s == nullptr) {
+        return false;
+    }
+    char *saveptr;
+    s += strlen(persistent_header);
+    for (char *p = strtok_r(s, "\n", &saveptr);
+         p; p = strtok_r(nullptr, "\n", &saveptr)) {
+        char *eq = strchr(p, int('='));
+        if (eq) {
+            *eq = 0;
```
[the remark this answers is not in the document]
Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/Util.cpp:590]: actually this is needed as the line below is doing strcmp and will fail.

File: libraries/AP_OpenDroneID/AP_OpenDroneID.cpp (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_OpenDroneID/AP_OpenDroneID.cpp:148]: removed

**Comment by dakejahl** [comment]
Move `sysid_this_mav()` to public in **libraries/GCS_MAVLink/GCS.h**

**Pull request merged**